packages/adblocker-content/adblocker.ts

```diff
@@ -146,8 +163,27 @@ export class DOMMonitor {
     window: Pick<Window, 'document'> & { MutationObserver?: typeof MutationObserver },
   ): void {
     if (this.observer === null && window.MutationObserver !== undefined) {
+      // An array of updated nodes in this timespan (or debounce session).
+      // This variable needs to be kept outside of mutation observer callback
+      // because the role of this array is to keep and accumulate the elements
+      // to be passed as an argument before debouncing completes its lifecycle.
+      let updatedNodes: Element[] = [];
+
+      const debouncedHandleUpdatedNodes = debounce(() => {
+        // Keep the reference to the object
+        const updatedNodesInSession = updatedNodes;
+
+        // Initialise and assign new object instead of modifying the existing one.
+        // Modifying the existing one will impact to the function running after this onComplete callback.
+        updatedNodes = [];
+
+        this.handleUpdatedNodes(updatedNodesInSession);
```

Comment:
cannot it be 
```
this.handleUpdatedNodes(updatedNodes);
updatedNodes = []
```
?

Reply:
Calling the function first then assigning new object later won't let us to make sure there is no change on the object.